### PR TITLE
[QC Check] Squish bug when vadr_num_alerts is a String

### LINKED ISSUE #243: [TheiaCoV] QC_Check task fails due to expecting int type on VADR num alerts
:bug:

### :pencil: Describe the Issue

From @emily-smith1 

> We are seeing failures at the qc_check in TheiaCov whenever VADR is skipped which is causing the whole workflow to fail. Not urgent, but might be something worth fixing for the next patch release 

### :repeat: How to Reproduce

- Was this run on the Terra platform?
  - Terra on GCP or Azure?
- Or were the workflows run locally using `miniwdl` or `cromwell`?
  - **What was the exact command was used to launch the workflow?**
- If possible, we may ask you to share your Terra workspace with us. Usually READER access is sufficient, but we may ask for WRITER access if we need to make changes to the workspace to reproduce the issue.

### PR DESCRIPTION
**Pull request by sage-wright**
This PR closes #243 .

🗑️ This dev branch should be deleted after merging to main.

## :brain: Aim, Context and Functionality

This PR puts the QC check of vadr_num_alerts into a try-except block and if failure, makes a QC_ALERT saying that it was not an integer and couldn't be compared (presumably because it was such a bad assembly)

## :hammer_and_wrench:  Impacted Workflows/Tasks & Changes Being Made
This will affect the behavior of the workflow(s) even if users don’t change any workflow inputs relative to the last version : Yes

It won't fail now.

Also, removed vestigial task and updated the pytest filter.

Running this workflow on different occasions could result in different results, e.g. due to use of a live database, "latest" docker image, or stochastic data processing : No 

#### 🔄 Data Processing 

Just added a try-except around the vadr_num_alerts block

Docker/software or software versions changed: 

Databases or database versions changed:

Data processing/commands changed:

File processing changed:

Compute resources changed:

#### ➡️ Inputs 

None

#### ⬅️ Outputs 

None

#### Terra Testing

- [x] testing on original failure case [here]( https://app.terra.bio/#workspaces/cdc-terra-resources/Theiagen_Wright_SC2_Sandbox/job_history/f7936e04-fb0b-4d89-abec-82579a4bfcec)

#### Suggested Scenarios for Reviewer to Test

Test by running the qc check part on a sample that has a string for vadr_num_alerts

## :microscope: Final Developer Checklist

- [x] The workflow/task has been tested locally and results, including file contents, are as anticipated
- [x] The workflow/task has been tested on Terra and results, including file contents, are as anticipated
- [x] The CI/CD has been adjusted and tests are passing (to be completed by Theiagen developer)
- [x] Code changes follow the [style guide](https://theiagen.notion.site/Style-Guide-WDL-Workflow-Development-bb456f34322d4f4db699d4029050481c)

## 🎯 Reviewer Checklist 

- [x] All impacted workflows/tasks have been tested on Terra with a different dataset than used for development
- [x] All reviewer-suggested scenarios have been tested and any additional
- [x] All changed results have been confirmed to be accurate
- [x] All workflows/tasks impacted by change/s have been tested using a standard validation dataset to ensure no unintended change of functionality
- [x] All code adheres to the style guide
- [x] MD5 sums have been updated
- [x] The PR author has addressed all comments

## 🗂️ Associated Documentation (to be completed by Theiagen developer)

- [ ] Relevant documentation on the Public Health Resources "PHB Main" has been updated
- [ ] Workflow diagrams have been updated to reflect changes

### CODE REVIEW

File: tests/config/pytest_filter.yml [hunk 1]
```diff
@@ -9,7 +9,7 @@ wf_theiacov_clearlabs:
   - tasks/taxon_id/contamination/task_kraken2.wdl
   - tasks/taxon_id/task_nextclade.wdl
   - tasks/species_typing/betacoronavirus/task_pangolin.wdl
-  - tasks/quality_control/basic_statistics/task_sc2_gene_coverage.wdl
+  - tasks/quality_control/basic_statistics/task_gene_coverage.wdl
```
Comment (kevinlibuit) [inline tests/config/pytest_filter.yml:12]: Is this cross talk from a different branch?

File: tests/config/pytest_filter.yml [hunk 2]
```diff
@@ -72,7 +72,7 @@ wf_theiacov_ont:
   - tasks/taxon_id/task_nextclade.wdl
   - tasks/species_typing/betacoronavirus/task_pangolin.wdl
   - tasks/species_typing/lentivirus/task_quasitools.wdl
-  - tasks/quality_control/basic_statistics/task_sc2_gene_coverage.wdl
+  - tasks/quality_control/basic_statistics/task_gene_coverage.wdl
```
Comment (kevinlibuit) [inline tests/config/pytest_filter.yml:75]: Same comment x3 as above RE cross-talk

Comment (sage-wright) [reply tests/config/pytest_filter.yml:75]: yikes, yes it is lol 😂

Comment (sage-wright) [reply tests/config/pytest_filter.yml:75]: should we keep or remove?

**Comment by kevinlibuit** [comment]
Changes look good! Not sure if, at least in `main` and your current dev branch, that the basic stat task file needs to be updated at this point.

Just launched my own [functional tests](https://app.terra.bio/#workspaces/theiagen-validations/Libui-Sandbox_2023/job_history/327ee964-e2f5-4c22-a12b-ae50af4b8353)

**Review by emmadoughty — APPROVED** [review]
[Test run](https://app.terra.bio/#workspaces/theiagen-training-workspaces/APHL_PRY_demo/job_history/dbd5ce1e-010b-4b99-8cfb-ad4f74fc9109) completed successfully with expected results populated to the Terra data table. Well done, Sage!

File: tests/config/pytest_filter.yml (hunk 2 above)
Comment (kevinlibuit) [reply tests/config/pytest_filter.yml:75]: Haha let's remove it. It will be a confusing conflict for other branches that may get PR'ed before #341

Comment (sage-wright) [reply tests/config/pytest_filter.yml]: removed!

**Review by kevinlibuit — APPROVED** [review]
Changes look great. Thanks for the quick bug squish, Sage!

For posterity's sake, I'm adding a note indicating that the vestigial task mentioned above is `/task_qc_check.wdl`: this task no longer used as its function is fully captured (and improved) in `/task_qc_check_phb.wdl`

**Pull request merged**